parcels/field.py

```diff
@@ -916,23 +917,39 @@ def add_periodic_halo(self, zonal, meridional, halosize=5, data=None):
         data = self.data if dataNone else data
         if zonal:
             if len(data.shape) is 3:
-                data = np.concatenate((data[:, :, -halosize:], data,
-                                       data[:, :, 0:halosize]), axis=len(data.shape)-1)
+                if hasattr(self.grid, 'ti') and self.grid.ti == -1:
```

Comment:
@delandmeterp, I have now used the statement `hasattr(self.grid, 'ti') and self.grid.ti == -1` to check if `Field.data` is fully loaded or not. However, this might not be the most compact/intuitive way to check that. 

Is there a more elegant way to confirm if the `Field` is fully loaded or not?